earth2studio/models/dx/climatenet.py

```diff
@@ -79,23 +79,60 @@ def __init__(
         self.register_buffer("center", center)
         self.register_buffer("scale", scale)
 
-    input_coords = OrderedDict(
-        {
-            "batch": np.empty(0),
-            "variable": np.array(VARIABLES),
-            "lat": np.linspace(90, -90, 721, endpoint=True),
-            "lon": np.linspace(0, 360, 1440, endpoint=False),
-        }
-    )
-
-    output_coords = OrderedDict(
-        {
-            "batch": np.empty(0),
-            "variable": np.array(OUT_VARIABLES),
-            "lat": np.linspace(90, -90, 721, endpoint=True),
-            "lon": np.linspace(0, 360, 1440, endpoint=False),
-        }
-    )
+    @property
+    def input_coords(self) -> CoordSystem:
+        """Input coordinate system of diagnostic model
+
+        Returns
+        -------
+        CoordSystem
+            Coordinate system dictionary
+        """
+        return OrderedDict(
+            {
+                "batch": np.empty(0),
+                "variable": np.array(VARIABLES),
+                "lat": np.linspace(90, -90, 721, endpoint=True),
+                "lon": np.linspace(0, 360, 1440, endpoint=False),
+            }
+        )
+
+    @batch_coords()
+    def output_coords(self, input_coords: CoordSystem | None = None) -> CoordSystem:
+        """Ouput coordinate system of diagnostic model
+
+        Parameters
+        ----------
+        input_coords : CoordSystem
+            Input coordinate system to transform into output_coords
+            by default None, will use self.input_coords.
+
+        Returns
+        -------
+        CoordSystem
+            Coordinate system dictionary
+        """
+
+        if input_coords is None:
+            return OrderedDict(
+                {
+                    "batch": np.empty(0),
+                    "variable": np.array(OUT_VARIABLES),
+                    "lat": np.linspace(90, -90, 721, endpoint=True),
+                    "lon": np.linspace(0, 360, 1440, endpoint=False),
+                }
+            )
+
+        handshake_dim(input_coords, "lon", 3)
```

Comment:
having the handshake methods in the output coords seems like a good call, should this be the expectation for all models?

Reply:
Yes, I am moving all of the handshakes to the `output_coords` method.